src/configuration.c

```diff
@@ -112,55 +159,36 @@ bool read_config_from_file(char *config_file, struct configuration *config)
                                         return false;
                                 }
                         } else {
-                                fprintf(stderr, "ERR: missing key with string value: %s\n", config_key_str[i]);
-                                return false;
+                                config->strValues[i] = strdup(config_str_default[i]);
                         }
+
                 }
-                for (int i = CONF_INT_MIN + 1; i < CONF_INT_MAX; i++) {
+
+                for (int i = CONF_INT_MIN; i < CONF_INT_MAX; i++) {
                         char *ptr;
                         ptr = nc_hashmap_get(nc_hashmap_get(keyfile, "settings"), config_key_int[i]);
                         if (ptr) {
                                 config->intValues[i] = strtoll(ptr, NULL, 10);
                         } else {
-                                fprintf(stderr, "ERR: missing key with integer value: %s\n", config_key_int[i]);
-                                return false;
+                                config->intValues[i] = config_int_default[i];
                         }
                 }
 
-                for (int i = CONF_BOOL_MIN + 1; i < CONF_BOOL_MAX; i++) {
+                for (int i = CONF_BOOL_MIN; i < CONF_BOOL_MAX; i++) {
                         char *ptr;
                         ptr = nc_hashmap_get(nc_hashmap_get(keyfile, "settings"), config_key_bool[i]);
-                        config->boolValues[i] = false;
-
                         if (ptr) {
-                                if (strcasecmp(ptr, "TRUE") == 0) {
-                                        config->boolValues[i] = true;
-                                }
-                                if (strcasecmp(ptr, "1") == 0) {
+                                if ((strcasecmp(ptr, "TRUE") == 0) || (strcasecmp(ptr, "1") == 0)) {
                                         config->boolValues[i] = true;
+                                } else if ((strcasecmp(ptr, "FALSE") == 0) || (strcasecmp(ptr, "0") == 0)) {
+                                        config->boolValues[i] = false;
                                 }
```

Comment:
So what happens if *ptr is not 0,1,TRUE or FALSE? Shouldn't be here an "else return false"?